.circleci/config.yml

```diff
@@ -55,3 +55,11 @@ jobs:
       - run:
           name: Project Tests
           command: yarn test:projects:cra-ts
+      - run:
+          name: Publish npm package
+          command: |
+            if [ -n "${npm_TOKEN+x}" ] && [ "${CIRCLE_BRANCH}" == "master" ]; then
+              echo "//registry.npmjs.org/:_authToken=${npm_TOKEN}" > ~/project/.npmrc
+              npm version prerelease --preid="${CIRCLE_BUILD_NUM}.${CIRCLE_BRANCH}.`git rev-parse --short HEAD`" --no-git-tag-version
```

Comment:
my initial thought was about not introducing any Circle CI dependencies to the version tag, but have a guess that you are doing this intentionally for the sake of providing some mechanism to transparently upgrade 'experimental' packages on the client side (and, thus, you do need some number that would be monotonically incremented in the version), right?

Reply:
still, according to the documentation, tags complements semantic versioning (in a sense that those serve as a substitution for version number) - so, the question is whether versioning will take effect for different packages published under the same tag? or the published package (with the same tag) will be just replaced in the NPM registry?